From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Mon, 5 Apr 2021 15:32:33 -0400
Subject: [PATCH] grc: Be more informative when missing "options" block.

This raises an RuntimeError describing that GRC could not find the
built-in blocks rather than raising a KeyError on "options", which
left users with little to go on. This error has come up in various
situations involving an incorrect blocks path (due to mixing GR
versions causing the configuration files to point to the wrong
installation) or simply the blocks not being installed.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 grc/core/platform.py | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/grc/core/platform.py b/grc/core/platform.py
index 46a795bd0..4c0ffc943 100644
--- a/grc/core/platform.py
+++ b/grc/core/platform.py
@@ -194,7 +194,20 @@ class Platform(Element):
 
         self._docstring_extractor.finish()
         # self._docstring_extractor.wait()
-        utils.hide_bokeh_gui_options_if_not_installed(self.blocks['options'])
+        if 'options' not in self.blocks:
+            # we didn't find one of the built-in blocks ("options")
+            # which probably means the GRC blocks path is bad
+            errstr = (
+                "Failed to find built-in GRC blocks (specifically, the "
+                "'options' block). Ensure your GRC block paths are correct "
+                "and at least one points to your prefix installation:"
+            )
+            errstr = "\n".join([errstr] + self.config.block_paths)
+            raise RuntimeError(errstr)
+        else:
+            # might have some cleanup to do on the options block in particular
+            utils.hide_bokeh_gui_options_if_not_installed(self.blocks['options'])
+
 
     def _iter_files_in_block_path(self, path=None, ext='yml'):
         """Iterator for block descriptions and category trees"""
-- 
2.30.2

